src/mempool_space/mod.rs

```diff
@@ -0,0 +1,47 @@
+pub mod api;
+pub mod websocket;
+
+use api::{BlockEvent, MempoolSpaceWebSocketRequestData, MempoolSpaceWebSocketRequestMessage};
+use bitcoin::Network;
+use futures_core::Stream;
+use url::Url;
+
+pub async fn subscribe_to_new_blocks(
+    network: &Network,
+    message: &MempoolSpaceWebSocketRequestMessage,
+) -> anyhow::Result<impl Stream<Item = BlockEvent>> {
+    let url: Url = url::Url::parse(&build_websocket_address(network)).unwrap();
+    websocket::connect_and_publish_message(url, message).await
+}
+
+pub fn build_websocket_request_message(
```

Comment:
This should be a private function for now.

Reply:
@LLFourn Good, I've updated it to be private and moved this one to websocket.rs as it's only used by websocket, wdyt ?